Reusable test framework the Inspector tests

Juntian777 opened the pull request:
Summary:
This Diff introduces a reusable test framework. The inspector_test_utils.py file provides methods to instantiate the model, retrieve expected outputs, and assert the correctness of actual outputs in a extensible way.

Also, update the intermediate_output_capturer_test to take advantage of the reusable framework and make the intermediate_output_capturer_test more extensible by using reusable setup method, making it easier to add new models and tests.

Differential Revision: D75803288

Gasoonjia reviewed (approved):
LGTM. Thanks for continue work

Commit:
Reusable test framework the Inspector tests (#11314)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/11314

This Diff introduces a reusable test framework. The inspector_test_utils.py file provides methods to instantiate the model, retrieve expected outputs, and assert the correctness of actual outputs in a extensible way.

Also, update the intermediate_output_capturer_test to take advantage of the reusable framework and make the intermediate_output_capturer_test more extensible by using reusable setup method, making it easier to add new models and tests.

Reviewed By: Gasoonjia

Differential Revision: D75803288
